File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l,r = 0, len(matrix) * len(matrix[0]) - 1

        while l <= r:
            mid_start = (r + l) // 2
            ls = mid_start // len(matrix[0])
            mid = mid_start % len(matrix[0])

            search = matrix[ls][mid]
            if search == target:
                return True
            elif search < target:
                l = mid_start + 1
            else:
                r = mid_start - 1

        return False
        

File: Data Structures & Algorithms/reverse-string/submission-1.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int mid = s.size() / 2;
        int n = s.size();
        for(int i = 0; i < mid; i++){
            swap(s[i], s[n - 1 - i]);
        }
    }
};